### PR TITLE
Add option to set string value quoting character

### PR DESCRIPTION
**Pull request by Creeperface01**
I ran into a problem when there were enforced double quotes in third party SQL processor.
I don't think the quote character matters in case of SQL server but still might be a good option here in some cases.

### CODE REVIEW

File: packages/react-querybuilder/src/defaults.ts [hunk 1]
```diff
@@ -42,6 +42,8 @@ export const defaultPlaceholderOperatorGroupLabel = placeholderLabel;
  */
 export const defaultJoinChar = ',';
 
+export const defaultQuoteChar: `'` | `"` = `'`;
```
Comment (jakeboone02) [inline packages/react-querybuilder/src/defaults.ts]: We don't need to export this or even define it at all. Just default it to `'` in `formatQuery`.

File: packages/react-querybuilder/src/types/importExport.ts [hunk 2]
```diff
@@ -128,6 +128,11 @@ export interface FormatQueryOptions {
    * @default '~'
    */
   placeholderOperatorName?: string;
+  /**
+   * Character to use for quoting string values in the SQL format.
+   * @default '''
+   */
+  quoteChar?: `'` | `"`;
```
Comment (jakeboone02) [inline packages/react-querybuilder/src/types/importExport.ts]: To align with the existing option `quoteFieldNamesWith`, let's name this one `quoteValuesWith`.

File: packages/react-querybuilder/src/utils/formatQuery/defaultValueProcessorByRule.ts [hunk 3]
```diff
@@ -1,24 +1,34 @@
...
 
-const escapeSingleQuotes = (
+const escapeStringValueQuotes = (
   v: string | number | boolean | object | null,
+  quoteChar: string,
   escapeQuotes?: boolean
-) => (escapeQuotes && typeof v === 'string' ? v.replaceAll(`'`, `''`) : v);
+) =>
+  escapeQuotes && typeof v === 'string'
+    ? v.replaceAll(`${quoteChar}`, `${quoteChar}${quoteChar}`)
```
Comment (jakeboone02) [inline packages/react-querybuilder/src/utils/formatQuery/defaultValueProcessorByRule.ts:11]: In your third-party SQL library, does a string like `"inner""quotes"` work the same as `'inner''quotes'` would in other SQL processors? Or does it require a different escape character like `\`?

File: packages/react-querybuilder/src/utils/formatQuery/formatQuery.SQL.test.ts [hunk 4]
```diff
@@ -566,3 +568,15 @@ describe('non-standard combinators', () => {
...
   });
 });
+
+it('Different quote character', () => {
+  expect(formatQuery(query, {
+    format: 'sql',
+    quoteChar: `'`,
+  })).toBe(sqlString);
+
+    expect(formatQuery(query, {
```
Comment (jakeboone02) [inline packages/react-querybuilder/src/utils/formatQuery/formatQuery.SQL.test.ts]: The inconsistent indentation here indicates you need to run prettier. You can run it on the entire codebase with `bun pretty-print`.

File: packages/react-querybuilder/src/utils/formatQuery/formatQuery.ts [hunk 5]
```diff
@@ -376,6 +384,7 @@ function formatQuery(ruleGroup: RuleGroupTypeAny, options: FormatQueryOptions |
           placeholderFieldName,
           placeholderOperatorName,
           validator,
+          quoteChar,
```
Comment (jakeboone02) [inline packages/react-querybuilder/src/utils/formatQuery/formatQuery.ts]: This option won't be used by parameterized rule processors, so it's not necessary to pass it here.

**Review by jakeboone02 — CHANGES_REQUESTED** [review]
Thanks for submitting this! I requested a few changes before kicking off CI.

I think this is a useful feature, although I've never seen a SQL dialect that required double quotes on values. How does it differentiate between identifiers and values? Other dialects use double quotes or backticks exclusively for identifiers and single quotes exclusively for values.

File: packages/react-querybuilder/src/types/importExport.ts (hunk 2 above)
Comment (jakeboone02) [reply packages/react-querybuilder/src/types/importExport.ts]: One more comment: type it as `string` to be as flexible as possible. If you really want to get fancy you can type it as `"'" | '"' | (string & {})`, which effectively types it as `string` but suggests `'` and `"` for autocomplete in IDEs like VS Code.

**Comment by jakeboone02** [comment]
@Creeperface01 are you still needing this feature? I'm inclined to merge this PR once my comments are addressed.

**Comment by Creeperface01** [comment]
> @Creeperface01 are you still needing this feature? I'm inclined to merge this PR once my comments are addressed.

Yes, sorry I was quite busy lately, but I'll do my best to address them asap

File: packages/react-querybuilder/src/utils/formatQuery/defaultValueProcessorByRule.ts (hunk 3 above)
Comment (Creeperface01) [reply packages/react-querybuilder/src/utils/formatQuery/defaultValueProcessorByRule.ts:11]: Yes it works the same. I mean it's not only about the processor but these quotes are supported by MySQL or MariaDB and they behave the same as single quotes afaik https://mariadb.com/kb/en/string-literals/

Comment (jakeboone02) [reply packages/react-querybuilder/src/utils/formatQuery/defaultValueProcessorByRule.ts:11]: Ok, that makes sense now. Thanks.

**Review by jakeboone02 — APPROVED** [review]

**Comment by jakeboone02** [comment]
@all-contributors please add @Creeperface01 for code, tests

**Comment by jakeboone02** [comment]
Thanks again! This will be available in the next release which should be out pretty quickly.

**Pull request merged**